### ISSUE TITLE
hover_card.trigger stops working when on_click event in rx.chakra.button in defined

### ISSUE THREAD

**Issue by Benjamin-Aw-93** [reflex-dev/reflex#2730]
**Describe the bug**
When implementing the on_click event in rx.chakra.button within the rx.hover_card.root, the hover_card.trigger stops working.

**To Reproduce**
The code is as follows:

```python
rx.hover_card.root(
    rx.hover_card.trigger(
        rx.chakra.button("Single Search", flex_grow="1", style={"background_color": "#2e5397", "color": "white"}, size="lg", on_click=my_on_click_handler)
    ),
    rx.hover_card.content(
        rx.text("Find the SSOC code for a single job title and job description", style={"font_weight": "bold", "color": "white"}),
        side="left",
        align="center",
        style={"background": "#C55B11"}
    ),
)
```

my_on_click_handler just redirects the user to another page.

```python
def my_on_click_handler():
    # Define what happens when the button is clicked
    return rx.redirect("/singlesearch")
```

**Expected behavior**
The hover_card.trigger should continue to work as intended, even when the on_click event is implemented in the rx.chakra.button. The hover_card.trigger should display the hover card content when the button is hovered over, and the on_click event of the button should trigger the specified my_on_click_handler function.

**Screenshots**
Ideally, the hover card should still appear like so:
![Screenshot 2024-02-27 153542](https://github.com/reflex-dev/reflex/assets/44826548/7267cad7-6fa4-46aa-95ec-2ace65f2fd66)

**Comment by Alek99** [comment]
Thanks for pointing this out we have a fix for this coming in the next release 

https://github.com/reflex-dev/reflex/pull/2793

**Comment by Benjamin-Aw-93** [comment]
Thank you! Will be looking forward to the next release

**Comment by picklelo** [comment]
Merged in #2793
